File: app/src/main/java/com/octavianmetta/android/myanimelistsearcher/rest/APIService.java
package com.octavianmetta.android.myanimelistsearcher.rest;

import com.octavianmetta.android.myanimelistsearcher.models.MALSearchResponse;
import com.octavianmetta.android.myanimelistsearcher.models.MALTopResponse;
import com.octavianmetta.android.myanimelistsearcher.models.anime.AnimeModel;


import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface APIService {

    String BASE_URL = "https://api.jikan.moe/v3/";

    @GET("search/{type}")
    Observable<MALSearchResponse> getSearch(@Path("type") String type,
                                            @Query("q") String title,
                                            @Query("page") Integer page);

    @GET("top/{type}/{page}/airing")
    Observable<MALTopResponse> getTopAnime(@Path("type") String type,
                                           @Path("page") Integer page);

    @GET("anime/{malId}")
    Observable<AnimeModel> getAnime(@Path("malId") Integer malId);
}

File: app/src/main/java/com/octavianmetta/android/myanimelistsearcher/rest/APIService.kt
package com.octavianmetta.android.myanimelistsearcher.rest

import com.octavianmetta.android.myanimelistsearcher.models.MALSearchResponse
import com.octavianmetta.android.myanimelistsearcher.models.MALTopResponse
import com.octavianmetta.android.myanimelistsearcher.models.anime.AnimeModel
import com.octavianmetta.android.myanimelistsearcher.models.anime.characterStaff.CharacterStaff


import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface APIService {

    @GET("search/{type}")
    fun getSearch(@Path("type") type: String,
                  @Query("q") title: String,
                  @Query("page") page: Int?,
                  @Query("limit") limit: Int?): Observable<MALSearchResponse>

    @GET("top/{type}/{page}/airing")
    fun getTopAnime(@Path("type") type: String,
                    @Path("page") page: Int?): Observable<MALTopResponse>

    @GET("anime/{malId}/")
    fun getAnime(@Path("malId") malId: Int?): Observable<AnimeModel>

    @GET("anime/{malId}/characters_staff")
    fun getCharacter(@Path("malId") malId: Int?): Observable<CharacterStaff>

    companion object {

        val BASE_URL = "https://api.jikan.moe/v3/"
    }
}
